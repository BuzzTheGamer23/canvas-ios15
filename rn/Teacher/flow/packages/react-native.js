// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
