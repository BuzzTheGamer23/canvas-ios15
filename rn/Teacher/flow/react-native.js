//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
