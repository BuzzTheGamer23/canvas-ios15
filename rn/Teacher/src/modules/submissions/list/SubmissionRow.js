// @flow

import React from 'react'
import {
  View,
  StyleSheet,
  TouchableHighlight,
  Image,
  Text,
} from 'react-native'
import DisclosureIndicator from '../../../common/components/DisclosureIndicator'
import Token from '../../../common/components/Token'
import i18n from 'format-message'
import type {
  GradeProp,
  SubmissionProp,
  SubmissionStatusProp,
} from './submission-prop-types'
import colors from '../../../common/colors'

type RowProps = {
  +testID: string,
  +onPress?: () => void,
  +children?: Array<any>,
  +disclosure?: boolean,

}
const Row = ({ onPress, testID, children, disclosure }: RowProps): * => {
  return (
    <View style={styles.row}>
      <TouchableHighlight style={styles.touchableHighlight} onPress={onPress} testID={testID}>
        <View style={styles.container}>
          {children}
          {disclosure ? <DisclosureIndicator /> : undefined}
        </View>
      </TouchableHighlight>
    </View>
  )
}

const SubmissionStatus = ({ status }: { status: SubmissionStatusProp }): * => {
  let color: string = '#8B969E' // none
  let title: string = i18n({
    default: 'No submission yet',
    description: 'No submission from the student for the given assignment',
  })

  switch (status) {
    case 'late':
      color = '#FC5E13'
      title = i18n({
        default: 'Late',
        description: 'Assignment was turned in late',
      })
      break
    case 'missing':
      color = '#EE0612'
      title = i18n({
        default: 'Missing',
        description: 'The assignment has not been turned in',
      })
      break
    case 'submitted':
      color = '#07AF1F'
      title = i18n({
        default: 'Submitted',
        description: 'The assignment has been turned in',
      })
      break
  }

  return <Text style={[styles.statusText, { color }]}>{title}</Text>
}

const Grade = ({ grade }: {grade: ?GradeProp}): * => {
  if (!grade) {
    return null
  }

  const ungraded = i18n({
    default: 'ungraded',
    description: 'Label for ungraded assignment submission',
  })

  return grade === 'ungraded'
    ? <Token style={{ alignSelf: 'center' }} color={ colors.primaryButton }>{ ungraded }</Token>
    : <Text style={[ styles.gradeText, { alignSelf: 'center' } ]}>{ grade }</Text>
}

const SubmissionRow = ({ userID, onPress, avatarURL, name, status, grade }: SubmissionProp): * => {
  return (
    <Row disclosure testID={`submission-${userID}`} onPress={onPress(userID)}>
      <Image source={{ uri: avatarURL }} style={styles.avatar} />
      <View style={styles.textContainer}>
        <Text
          style={styles.title}
          ellipsizeMode='tail'
          numberOfLines={2}>{name}</Text>
        <SubmissionStatus status={status} />
      </View>
      <Grade grade={grade} />
    </Row>
  )
}

export default SubmissionRow

const styles = StyleSheet.create({
  row: {
    flex: 1,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'lightgrey',
  },
  touchableHighlight: {
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'flex-start',
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 16,
    paddingRight: 16,
  },
  textContainer: {
    flex: 1,
    paddingLeft: 8,
  },
  title: {
    flex: 1,
    fontSize: 16,
    fontWeight: '600',
    color: '#2D3B45',
  },
  statusText: {
    fontSize: 14,
    paddingTop: 2,
  },
  gradeText: {
    fontSize: 14,
    color: colors.primaryButton,
  },
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 20,
    alignSelf: 'center',
  },
})
