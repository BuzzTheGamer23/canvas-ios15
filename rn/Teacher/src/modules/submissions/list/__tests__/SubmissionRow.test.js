// @flow

import 'react-native'
import React from 'react'
import SubmissionRow from '../SubmissionRow'
import type {
  SubmissionProp,
  SubmissionStatusProp,
  GradeProp,
} from '../submission-prop-types'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const mockSubmission = (status: SubmissionStatusProp = 'none', grade: ?GradeProp = null): SubmissionProp => {
  return {
    onPress: () => {},
    userID: '1',
    avatarURL: 'https://cats.pajamas/',
    name: 'Green Latern',
    status,
    grade,
  }
}

test('unsubmitted ungraded row renders correctly', () => {
  const submission = mockSubmission()

  let tree = renderer.create(
    <SubmissionRow {...submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('missing ungraded row renders correctly', () => {
  const submission = mockSubmission('missing')
  let tree = renderer.create(
    <SubmissionRow {...submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('late graded row renders correctly', () => {
  const submission = mockSubmission('late', 'B-')
  let tree = renderer.create(
    <SubmissionRow {...submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('submitted ungraded row renders correctly', () => {
  const submission = mockSubmission('submitted', 'ungraded')
  let tree = renderer.create(
    <SubmissionRow {...submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
