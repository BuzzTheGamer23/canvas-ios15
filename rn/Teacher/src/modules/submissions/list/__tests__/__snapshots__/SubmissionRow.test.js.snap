// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`late graded row renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
        },
      ]
    }
    testID="submission-1"
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cats.pajamas/",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Object {
              "color": "#2D3B45",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              Object {
                "color": "#FC5E13",
              },
            ]
          }
        >
          Late
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#368BD8",
              "fontSize": 14,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        B-
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`missing ungraded row renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
        },
      ]
    }
    testID="submission-1"
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cats.pajamas/",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Object {
              "color": "#2D3B45",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              Object {
                "color": "#EE0612",
              },
            ]
          }
        >
          Missing
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`submitted ungraded row renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
        },
      ]
    }
    testID="submission-1"
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cats.pajamas/",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Object {
              "color": "#2D3B45",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              Object {
                "color": "#07AF1F",
              },
            ]
          }
        >
          Submitted
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        color="#368BD8"
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "fontSize": 11,
              "fontWeight": "600",
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#368BD8",
              "color": "#368BD8",
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        UNGRADED
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`unsubmitted ungraded row renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
        },
      ]
    }
    testID="submission-1"
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cats.pajamas/",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Object {
              "color": "#2D3B45",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Green Latern
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 14,
                "paddingTop": 2,
              },
              Object {
                "color": "#8B969E",
              },
            ]
          }
        >
          No submission yet
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 2,
            "borderTopWidth": 2,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
